### PR TITLE
Implement #295: Add `loaders_from_representation` to api

### PR DESCRIPTION
**Pull request by BigRoy**
Implementation as descried in https://github.com/getavalon/core/issues/295 plus the accompanied cleanup across the tools.

Let me know if I missed anything.

### CODE REVIEW

File: avalon/pipeline.py [hunk 1]
```diff
@@ -840,4 +860,30 @@ def get_representation_path(representation):
...
+    has_representation = ("*" in Loader.representations or
+                          representation["name"] in Loader.representations)
+    return has_family and has_representation
+
+
+def loaders_from_representation(loaders, representation):
+    """Return all compatible loaders for a representation."""
+
+    context = get_representation_context(representation)
+    return [l for l in loaders if is_compatible_loader(l, context)]
```
[the remark this answers is not in the document]
Comment (BigRoy) [reply avalon/pipeline.py]: Interesting. @mottosso with https://github.com/getavalon/core/pull/297/commits/7068d5f3a8f977e9e9a22181e904fe4393eda1ea shouldn't this be passing?

[the remark this answers is not in the document]
Comment (mottosso) [reply avalon/pipeline.py]: It doesn't look like there is a newline at the end of the file?

File: avalon/pipeline.py [hunk 2]
```diff
@@ -840,4 +860,30 @@ def get_representation_path(representation):
         "subset": subset["name"],
         "version": version_["name"],
         "representation": representation["name"],
+        "user": getpass.getuser(),
```
Comment (mottosso) [inline avalon/pipeline.py]: Let's let this one be overridden by `AVALON_USER` as well.

**Comment by mottosso** [comment]
Ok, I think this looks really clean. Well done. :) I'm not in a position to test it today, but it you're happy with it then I am too.

File: avalon/pipeline.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mottosso) [reply avalon/pipeline.py]: https://github.com/getavalon/core/blob/7068d5f3a8f977e9e9a22181e904fe4393eda1ea/avalon/pipeline.py#L889

**Comment by BigRoy** [comment]
@mottosso @aardschok did you still see any issues with this? Can we merge this?

**Pull request merged**